src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -1276,149 +1373,55 @@ internal enum ParsingStatus
         }
 
         [DoesNotReturn]
-        internal static void ThrowOverflowOrFormatException(ParsingStatus status, ReadOnlySpan<char> value, TypeCode type = 0) => throw GetException(status, value, type);
-
-        [DoesNotReturn]
-        internal static void ThrowOverflowOrFormatException<TInteger>(ParsingStatus status, ReadOnlySpan<char> value)
+        internal static void ThrowOverflowOrFormatException<TChar, TInteger>(ParsingStatus status, ReadOnlySpan<TChar> value)
+            where TChar : unmanaged, IUtfChar<TChar>
             where TInteger : unmanaged, IBinaryIntegerParseAndFormatInfo<TInteger>
         {
-            throw GetException<TInteger>(status, value);
+            if (status == ParsingStatus.Failed)
+            {
+                ThrowFormatException(value);
+            }
+            ThrowOverflowException<TInteger>();
         }
 
         [DoesNotReturn]
-        internal static void ThrowOverflowException(TypeCode type) => throw GetOverflowException(type);
```

Comment:
Why remove this overload? It was added specifically to get smaller code size for a lot of methods.

Reply:
Because the handling is now done in a different/better way that still keeps them small.